server/src/models/user.rs

```diff
@@ -1,10 +1,27 @@
+use chrono::Utc;
 use diesel::prelude::*;
+use uuid::Uuid;
+use serde::{Serialize, Deserialize};
 
-#[derive(Queryable)]
+use crate::schema::users;
+
+#[derive(Serialize, Deserialize, Queryable, Insertable)]
 pub struct User {
 	pub id: String,
-	pub username: String,
 	pub created_at: String,
+	pub username: String,
 	pub balance: i32,
-	pub role: i32
+	pub role: String
+}
+
+impl User {
+		pub fn new(username: String, balance: i32, role: String) -> Self {
+			User {
+				id: Uuid::new_v4().to_string(),
+				created_at: Utc::now().to_rfc3339(),
+				username,
+				balance,
+				role
```

Comment:
Can we set role to be the enum or does that not work with our SQL library? If not, the constructor could take an enum probably.

Reply:
Opened #4